modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImpl.java

```diff
@@ -563,6 +537,91 @@ private Filter _getORFilter(Filter leftFilter, Filter rightFilter) {
 		return booleanFilter;
 	}
 
+	private Optional<Object> _getResultVisitBinaryExpressionOptional(
+		BinaryExpression.Operation operation, Object left, Object right,
+		Locale locale) {
+
+		Object resultVisitBinaryExpression = null;
+
+		if (Objects.equals(BinaryExpression.Operation.AND, operation)) {
+			resultVisitBinaryExpression = _getANDFilter(
+				(Filter)left, (Filter)right);
+		}
+		else if (Objects.equals(BinaryExpression.Operation.EQ, operation)) {
+			resultVisitBinaryExpression = _getEQFilter(
+				(EntityField)left, right, locale);
+		}
+		else if (Objects.equals(BinaryExpression.Operation.GE, operation)) {
+			resultVisitBinaryExpression = _getGEFilter(
+				(EntityField)left, right, locale);
+		}
+		else if (Objects.equals(BinaryExpression.Operation.GT, operation)) {
+			resultVisitBinaryExpression = _getGTFilter(
+				(EntityField)left, right, locale);
+		}
+		else if (Objects.equals(BinaryExpression.Operation.LE, operation)) {
+			resultVisitBinaryExpression = _getLEFilter(
+				(EntityField)left, right, locale);
+		}
+		else if (Objects.equals(BinaryExpression.Operation.LT, operation)) {
+			resultVisitBinaryExpression = _getLTFilter(
+				(EntityField)left, right, locale);
+		}
+		else if (Objects.equals(BinaryExpression.Operation.NE, operation)) {
+			resultVisitBinaryExpression = _getNEFilter(
+				(EntityField)left, right, locale);
+		}
+		else if (Objects.equals(BinaryExpression.Operation.OR, operation)) {
+			resultVisitBinaryExpression = _getORFilter(
+				(Filter)left, (Filter)right);
+		}
+		else if (Objects.equals(BinaryExpression.Operation.SUB, operation)) {
+			resultVisitBinaryExpression = _getSUB(left, right);
+		}
+		else {
+			return Optional.empty();
+		}
+
+		return Optional.of(resultVisitBinaryExpression);
+	}
+
+	private Object _getSUB(Object left, Object right) {
```

Comment:
This name doesn't follow the rest of the class names, it should say _getSubWhatIsTheResultOfTheSub or _sub.

Reply:
Ok, I change it to _sub.